pkg/trace/remoteconfighandler/remote_config_handler_test.go

```diff
@@ -229,3 +229,126 @@ func TestLogLevel(t *testing.T) {
 
 	ctrl.Finish()
 }
+
+func TestStartWithMRF(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	mrfClient := NewMockRemoteClient(ctrl)
+	agentConfig := config.AgentConfig{
+		RemoteConfigClient:    remoteClient,
+		MRFRemoteConfigClient: mrfClient,
+		DebugServerPort:       1,
+	}
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+	pkglog.SetupLogger(pkglog.Default(), "debug")
+
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	remoteClient.EXPECT().Subscribe(state.ProductAPMSampling, gomock.Any()).Times(1)
+	remoteClient.EXPECT().Subscribe(state.ProductAgentConfig, gomock.Any()).Times(1)
+	remoteClient.EXPECT().Start().Times(1)
+	mrfClient.EXPECT().Subscribe(state.ProductAgentFailover, gomock.Any()).Times(1)
+	mrfClient.EXPECT().Start().Times(1)
+
+	h.Start()
+
+	ctrl.Finish()
+}
+
+func TestMRFUpdateCallback(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	remoteClient := NewMockRemoteClient(ctrl)
+	mrfClient := NewMockRemoteClient(ctrl)
+	prioritySampler := NewMockprioritySampler(ctrl)
+	errorsSampler := NewMockerrorsSampler(ctrl)
+	rareSampler := NewMockrareSampler(ctrl)
+	pkglog.SetupLogger(pkglog.Default(), "debug")
+
+	agentConfig := config.AgentConfig{
+		RemoteConfigClient:    remoteClient,
+		MRFRemoteConfigClient: mrfClient,
+		DebugServerPort:       1,
+	}
+	h := New(&agentConfig, prioritySampler, rareSampler, errorsSampler)
+
+	// Test enabling MRF
+	enableAPM := true
+	mrfConfig := map[string]interface{}{
+		"failover_apm": &enableAPM,
```

Comment:
looks like there's no need for a variable here? (or other similar tests)
```suggestion
		"failover_apm": true,
```